csrc/ir/nodes.cpp

```diff
@@ -2063,6 +2063,32 @@ void MmaOp::setMacro(MmaMacro macro) {
   attribute<MmaMacro>(ATTR_POS_MACRO) = macro;
 }
 
+std::vector<PolymorphicValue> MmaOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+  // MmaOp is preceded by broadcast. For aten::matmul use the non-broadcasted
+  // inputs.
+  const auto& a =
+      ee.evaluate(input(0)->definition()->input(0)).as<at::Tensor>();
+  const auto& b =
+      ee.evaluate(input(1)->definition()->input(0)).as<at::Tensor>();
```

Comment:
You should check that these definitions are non-null and that they are `BroadcastOp` if that is what is assumed. There is discussion now whether it will always be the case that MmaOp will always be preceded by `BroadcastOp` as the interface is evolving.

Reply:
It seems more robust to me to instead try and squeeze and matmul without pattern matching the definitions of `MmaOp` inputs. For max generality this could be done with https://github.com/pytorch/pytorch/blob/8ff55c7e689d4c6f89f806112e04ff56dbfb6a3f/aten/src/ATen/native/Linear.cpp#L137 but I think that is not exposed. `einsum` is though, if you wanted to go that route.